lib/dragonfly_libvips/dimensions.rb

```diff
@@ -14,8 +14,18 @@ def call
           OpenStruct.new(width: orig_w, height: orig_h, scale: 1)
         when fill_area?
           OpenStruct.new(width: fill_width, height: fill_height)
-        when crop?
-          OpenStruct.new(width: width, height: height, x: xoffset, y: yoffset, scale: scale)
+        when crop_with_gravity?
+          OpenStruct.new(resize_width: fill_width,
+                         resize_height: fill_height,
+                         width: dimensions.width,
+                         height: dimensions.height,
+                         x: crop_gravity_x_offset,
+                         y: crop_gravity_y_offset)
+        when crop_without_gravity?
+          OpenStruct.new(width: dimensions.width,
+                         height: dimensions.height,
+                         x: x_offset,
+                         y: y_offset)
```

Comment:
We need to split these two, as cropping with gravity (i.e. `300x300#c`) first resizes the image and then crops, while cropping without (i.e. `300x300+50+50`) doesn't resize the image, but just grabs the portion of the original image.